src/Syntax/Term.hs

```diff
@@ -97,6 +100,32 @@ cata var alg = go where
     Var v -> var v
     Alg t -> alg (hmap go t)
 
+cataM
+  :: forall sig m f a
+  .  ( HTraversable sig
+     , Monad m
+     )
+  => (forall x . x -> m (f x))
+  -> (forall x . sig f x -> m (f x))
+  -> (Term sig a -> m (f a))
+cataM var alg = go where
+  go :: forall a . Term sig a -> m (f a)
+  go = \case
+    Var v -> var v
+    Alg t -> alg =<< htraverse go t
+
+handle
+  :: forall syn sig m a
+  .  ( HTraversable sig
+     , HTraversable syn
+     , Monad m
+     )
+  => (forall   x . x       -> m (Term sig x))
+  -> (forall t x . syn t x -> m (Term sig x))
+  -> Term (syn :+: sig) a
+  -> m (Term sig a)
+handle var alg = cataM var (unSum alg (pure . Alg))
```

Comment:
Per the above, defining `handle` in terms of `cataM` enables us to avoid requiring a `Traversable m` constraint.